src/bio/bio_xstream.c

```diff
@@ -1626,9 +1715,12 @@ bio_nvme_poll(struct bio_xs_context *ctxt)
 	 * Query and print the SPDK device health stats for only the device
 	 * owner xstream.
 	 */
-	if (ctxt->bxc_blobstore != NULL &&
-	    is_bbs_owner(ctxt, ctxt->bxc_blobstore))
-		bio_bs_monitor(ctxt, now);
+	for (st = SMD_DEV_TYPE_DATA; st < SMD_DEV_TYPE_MAX; st++) {
+		bxb = ctxt->bxc_xs_blobstores[st];
+		if (bxb && bxb->bxb_blobstore &&
+		    is_bbs_owner(ctxt, bxb->bxb_blobstore))
+			bio_bs_monitor(ctxt, st, now);
```

Comment:
If WAL, META and DATA share same blobstore, this bio_bs_monitor() against same device will be called three times, I suggest we don't do health monitor for META & WAL device for this moment, so that we can minimize the code changes and focus on I/O functionality.

Reply:
nm, 'bxb' == NULL is checked, so the bio_bs_monitor() won't be called multiple times against same device.